src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,83 @@
+package parser;
+
+import commands.*;
+import exceptions.SnomException;
+
+import java.util.Arrays;
+
+public class Parser {
+
+    /**
+     * Returns a command child class based on the given string of command.
+     *
+     * @param userInput     string of user input
+     * @return               Command child classes
+     * @throws SnomException if command is unknown
+     */
+    public static Command parse(String userInput) throws SnomException {
+        String commandStr = parseCommandStr(userInput);
+        String commandContent = parseCommandContent(userInput);
+        CommandEnum commandEnum = CommandEnum.getCommand(commandStr);
+        switch(commandEnum){
+        case LIST:
+            return new ListCommand(CommandEnum.LIST, commandContent);
+        case FINISH:
+            return new FinishCommand(CommandEnum.FINISH, commandContent);
+        case DELETE:
+            return new DeleteCommand(CommandEnum.DELETE, commandContent);
+        case BYE:
+            return new ExitCommand(CommandEnum.BYE, commandContent);
+        case TODO:
+            return new AddCommand(CommandEnum.TODO, commandContent);
+        case DEADLINE:
+            return new AddCommand(CommandEnum.DEADLINE, commandContent);
+        case EVENT:
+            return new AddCommand(CommandEnum.EVENT, commandContent);
+        default:
+            throw new SnomException("OOPS!!! I'm sorry, but I don't know what \'" + commandStr + "\' means :-(");
+        }
+    }
+
+    /**
+     * Returns the command string from user input
+     *
+     * @param userInput user input
+     * @return          command as string
+     */
+    public static String parseCommandStr(String userInput){
+        return userInput.split(" ")[0];
+    }
+
+    /**
+     * Returns the content of the command from user input
+     *
+     * @param userInput user input
+     * @return          content of the command
+     */
+    public static String parseCommandContent(String userInput) {
+        String[] splittedInputs = userInput.split(" ", 2);
+        if(splittedInputs.length >= 2){
+            return splittedInputs[1];
+        }else{
```

Comment:
Your code is overall really clean and neat so there is nothing much to comment on. But if I'm being super nit-picky, if I'm not wrong, I think the past tense of split is still split so splitted might cause some confusion? If not there is really nothing much to say :) Great Job !!!!